app/javascript/components/history/History.js

```diff
@@ -1,31 +1,134 @@
 import React from 'react';
 import { PropTypes } from 'prop-types';
-import { Card } from 'react-bootstrap';
+import { Button, Card } from 'react-bootstrap';
+import axios from 'axios';
+import reportError from '../util/ReportError';
+import moment from 'moment-timezone';
 
 import { formatTimestamp } from '../../utils/DateTime';
 import moment from 'moment-timezone';
 import { time_ago_in_words } from './helpers';
 
-const history = ({ history }) => {
-  return (
-    <Card className="card-square mt-4 mx-3 shadow-sm">
-      <Card.Header>
-        <b>{history.created_by}</b>, {time_ago_in_words(moment(history.created_at).toDate())} ago ({formatTimestamp(history.created_at)})
-        <span className="float-right">
-          <div className="badge-padding h5">
-            <span className="badge badge-secondary">{history.history_type}</span>
-          </div>
-        </span>
-      </Card.Header>
-      <Card.Body>
-        <Card.Text>{history.comment}</Card.Text>
-      </Card.Body>
-    </Card>
-  );
-};
+class History extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      edit_mode: false,
+      loading: false,
+      comment: '',
+    };
+  }
+
+  handleEditComment = () => {
+    this.setState({ edit_mode: true });
+  };
+
+  handleTextChange = event => {
+    this.setState({ comment: event.target.value });
+  };
+
+  handleDeleteComment = () => {
```

Comment:
I just noticed that there is no confirmation for delete... I think there needs to be a confirmation when you click the delete button

Reply:
You can use the confirmDialog component, would be super simple